File: Ejercicios iOS/Autos-Objective-C/AutoAppp/Auto.m
//
//  Auto.m
//  AutoAppp
//
//  Created by Virtualizacion02 on 28/08/23.
//

#import "Auto.h"

@implementation Auto

-(id) initWithMarca: (NSString *) marca Modelo: (NSString *)modelo{
    self = [super self];
    
    _marca = marca;
    _modelo = modelo;
    _x = 0;
    
    return self;
}

-(void) avanzar{
    
    _x++;
}

-(void) avanzarWithDistancia: (int)km{
    
    _x += km;
}

-(NSString *) imprimir{
    
    return [NSString stringWithFormat:@"%@ %@ %d",_marca,_modelo,_x];
}

@end

File: Ejercicios iOS/Autos-Swift/AutosSwift/Auto.swift
//
//  Auto.swift
//  AutosSwift
//
//  Created by Virtualizacion02 on 28/08/23.
//

import UIKit

class Auto: NSObject {
    
    let marca:String
    let modelo:String
    var km:Int
    
    init(marca: String, modelo: String) {
        self.marca = marca
        self.modelo = modelo
        self.km = 0
    }
    
    func imprimir()->String{
        var imprimir:String = "\(marca)  \(modelo) \(km)"
        return imprimir
    }
    
    func avanzarWithDistancia(){
        self.km += 10
    }

    

}